packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1224,6 +1224,35 @@ export class KeyringController extends BaseController<
     return await keyring.signUserOperation(address, userOp, executionContext);
   }
 
+  /**
+   * Changes the password to encrypt the current vault.
+   *
+   * @param password - The new password to encrypt with.
+   * @returns Promise resolving when the operation completes.
+   */
+  setPassword(password: string): Promise<void> {
+    return this.#persistOrRollback(async () => {
+      if (!this.state.isUnlocked || !this.#password) {
```

Comment:
Hmm, why are we requiring that `#password` be set here? This would prevent changing the password in a session where the vault was unlocked with an encryption key.

Reply:
Either way, we should probably have a test case for this scenario as well

Reply:
Ah, good point! `!this.state.isUnlocked` should be enough as a condition. There is already a case that tests `setPassword` while the controller is locked

Reply:
Done [66dc2cb](https://github.com/MetaMask/core/pull/4279/commits/66dc2cb96b9942ca84b6adc38ad8f63f7bd2a388)